ansible/roles/gpu/files/Dockerfile.ubuntu

```diff
@@ -0,0 +1,7 @@
+FROM ubuntu@sha256:82becede498899ec668628e7cb0ad87b6e1c371cb8a1e597d83a47fac21d6af3
+RUN apt update && apt install gpg curl -y
+RUN distribution=$(. /etc/os-release;echo $ID$VERSION_ID) \
+    && curl -s -L https://nvidia.github.io/nvidia-docker/gpgkey |  apt-key add - \
+    && curl -s -L https://nvidia.github.io/nvidia-docker/$distribution/nvidia-docker.list | tee /etc/apt/sources.list.d/nvidia-docker.list
+
+RUN  apt-get update &&  apt-get install -y nvidia-container-runtime=3.5.0-1 -V
```

Comment:
https://github.com/mesosphere/konvoy-image-builder/pull/77/commits/8e0503095f74a9f3198e86255d492af5b35ad40b#r698767005

Reply:
This one is still confusing to me as the nvidia-container-toolkit is what supersedes the nvidia-container-runtime-hook according to https://gitlab.com/nvidia/container-toolkit/container-toolkit/-/blob/master/packaging/debian/changelog#L106

Reply:
Right. It needs both the nvidia-container-runtime and  nvidia-container-toolkit packages. The toolkit package is a dependencyh of the runtime package, so by installing runtime, you also get toolkit.

The runtime package contains /usr/bin/nvidia-container-runtime which is the wrapper around runc that attempts to run the hook. Without it, there's no way to configure containerd to use the nvidia hook.

The hook is in the toolkit package which is symlinked to /usr/bin/nvidia-container-runtime-hook as is required in the code linked by Faiq, above.

Reply:
Thanks @joejulian that makes sense. I started a branch that was building the toolkit from the sources last weekend -- but maybe this ubuntu trickery does the job for now.

In any case, for posterity: https://github.com/mesosphere/konvoy-image-builder/commits/till/flatcar-oci